.github/workflows/rust.yml

```diff
@@ -26,12 +25,71 @@ jobs:
       uses: arduino/setup-protoc@v2
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
-    - name: Run tests
-      run: cargo test --workspace
-
-#   build:
-#     runs-on: ubuntu-latest
-#     steps:
-#     - uses: actions/checkout@v4
-#     - name: Build
-#       run: docker build . --tag=qdrant
+    - name: Install nextest
+      uses: taiki-e/install-action@nextest
+    - name: Build and run tests
+      # Profile "ci" is configured in .config/nextest.toml
+      run: cargo nextest run --workspace --profile ci
+    - name: Upload test report
+      uses: actions/upload-artifact@v3
+      with:
+        name: junit-${{ matrix.os }}.xml
+        path: target/nextest/ci/junit.xml
+        
+  # After tests are run, this hacky script will process the JUnit output of nextest
+  # and will create a GH Issue if there is a test marked as flaky
+  process-results:
+    runs-on: ubuntu-latest
```

Comment:
Could we incorporate this in the test job? That would prevent spawning 3 extra jobs just to process test failures.

Reply:
Unfortunately the script doesn't work well in windows

Reply:
I see.

In that case, lets experiment with this set up and see how we like it. Maybe we can improve upon this in the future if it becomes a problem.